map/bookmark_manager.hpp

```diff
@@ -161,8 +161,11 @@ class BookmarkManager final
     void SetCategoryAccessRules(kml::MarkGroupId categoryId, kml::AccessRules accessRules);
     void SetCategoryCustomProperty(kml::MarkGroupId categoryId, std::string const & key,
                                    std::string const & value);
-    bool DeleteBmCategory(kml::MarkGroupId groupId);
-
+    
+    /// Removes the category from the list of categories and deletes the related file.
+    /// - Parameters:
+    ///   - permanently: If true (default value), the file will be removed from the disk. If false, the file will be set as deleted.
+    bool DeleteBmCategory(kml::MarkGroupId groupId, bool permanently = true);
```

Comment:
Let's require an explicit `permanently` specification for callers for safety.
```suggestion
    bool DeleteBmCategory(kml::MarkGroupId groupId, bool permanently);
```

Reply:
This will require changing the Android code...

Reply:
Fixed!

Reply:
@rtsisyk to clearly specify the needed behavior in all places of usage.